### PR TITLE
- replaced mysqli with PDO

### PR DESCRIPTION
**Pull request by panzerfaust1**
- fixed DB\AQ not loading
- added sanity check to Page::metaJS()
- added DB\Adapter interface
- changed DB to be more PSR-12 friendly

### CODE REVIEW

File: src/Core/DB/MySQL.php [hunk 1]
```diff
@@ -1,137 +1,195 @@
...
-//                }
-                while ($row = mysqli_fetch_assoc($r)) {
-                    $rows[$row[$field1]] = $row;
-                }
-            } elseif ($field2) {
-                while ($row = mysqli_fetch_assoc($r)) {
-                    $rows[$row[$field1]][$row[$field2]] = $row;
-                }
+        if ($result === false) {
+            if ($q) {
```
Comment (glushkovds) [inline src/Core/DB/MySQL.php:126]: Must be deleted

File: src/Core/DB/Adapter.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    /**
+     * Makes custom assoc-array
+     *
+     * @param mixed $result Adapter-specific result object
+     * @param bool|string $f1 Column name to use as key or false to use index
+     * @param bool|string $f2 Second column name for inner rows
+     * @param bool|string $q Debug message to print if fails or false to not care
+     * @return mixed Array containing results
+     */
+    public function assoc(&$result, $f1, $f2, $q);
```
Comment (glushkovds) [inline src/Core/DB/Adapter.php:65]: delete $q

File: src/Core/DB.php [hunk 3]
```diff
@@ -2,173 +2,326 @@
...
+
         return false;
     }
 
+    /**
+     * @see Adapter::query()
+     * @param $q
+     * @return mixed
+     */
     public static function &query($q)
```
Comment (glushkovds) [inline src/Core/DB.php:66]: add PDO params with default empty array

File: src/Core/DB.php [hunk 4]
```diff
@@ -2,173 +2,326 @@
...
+
+            // total execution time
+            echo number_format($sumTime, $length);
+            echo '</td>';
             echo '<td style="white-space:nowrap;color:#666;padding:2px 4px">Общее время на запросы</td>';
             echo '</tr>';
             echo '</table>';
             echo '</div>';
 
             if (Core::uri(0) != 'admin') {
```
Comment (glushkovds) [inline src/Core/DB.php:241]: if (SF_LOCATION_ADMIN == SF_LOCATION) {

File: src/Core/DB.php [hunk 5]
```diff
@@ -2,173 +2,326 @@
...
+    public static function enumValues(string $table, string $field): array
     {
         $buffer = &$_ENV['enum_values'][$table][$field];
 
         if (!isset($buffer)) {
-            $q = "SHOW FULL COLUMNS FROM `$table` LIKE '$field'";
-            $row = DB::result($q);
-            $enumArray = array();
+            $row = DB::result("SHOW FULL COLUMNS FROM `$table` LIKE '$field'");
+
```
Comment (glushkovds) [inline src/Core/DB.php:309]: delete empty nl

File: src/Core/DB.php [hunk 6]
```diff
@@ -178,50 +331,80 @@ public static function enumValues($table, $field)
...
     {
         return static::$db->affectedRows();
     }
 
+    /**
+     * Starts DB transaction
+     */
     public static function transactionStart()
     {
         static::query('BEGIN');
```
Comment (glushkovds) [inline src/Core/DB.php:360]: refactor to PDO transactions

File: src/Core/DB/MySQL.php [hunk 7]
```diff
@@ -1,137 +1,195 @@
...
+    /**
+     * @param bool|PDOStatement $result
+     * @param int|string $field
+     *
+     * @return bool|array
+     */
+    public function result($result, $field = '')
     {
-        if (!$r) {
+        if ($result === false) {
```
Comment (glushkovds) [inline src/Core/DB/MySQL.php:101]: roolback

**Review by glushkovds — CHANGES_REQUESTED** [review]

**Comment by panzerfaust1** [comment]
- added requested changes

File: src/Core/DB/MySQL.php [hunk 8]
```diff
@@ -53,15 +53,15 @@ public function bind(array $vars)
 
     /**
      * @param string $q
-     *
+     * @param array $vars
```
Comment (glushkovds) [inline src/Core/DB/MySQL.php:56]: $vars rename to $params

**Pull request merged**